src/oidc/index.ts

```diff
@@ -0,0 +1,78 @@
+import { UserManager, WebStorageStateStore } from 'oidc-client-ts';
+
+type OidcConfiguration = {
+    issuer: string;
+    redirect_uri: string;
+    client_id: string;
+    response_type: string;
+    scope: string;
+    post_logout_redirect_uri: string;
+};
+
+/**
+ * Fetches the OIDC configuration for the given serverUrl.
+ *
+ * @param {string} [serverUrl] - The URL of the server to fetch the OIDC configuration from.
+ *                              If not provided, the value stored in local storage under the key 'config.server_url'
+ *                              is used.
+ *
+ * @returns {Promise<object>} - A promise resolving to the OIDC configuration.
+ *
+ * @throws {Error} - If there is a failure while fetching the OIDC configuration.
+ */
+export const fetchOidcConfiguration = async (): Promise<OidcConfiguration> => {
+    const serverUrlLocalStorage = localStorage.getItem('config.server_url');
+    const oidcUrl = `https://${serverUrlLocalStorage}/.well-known/openid-configuration`;
```

Comment:
Don't we need any check for `serverUrlLocalStorage`. What if it's `null` or empty?